src/Strategy/Cargo.hs

```diff
@@ -156,6 +201,69 @@ instance ToJSON CargoProject
 instance AnalyzeProject CargoProject where
   analyzeProject _ = getDeps
 
+data CargoPackage = CargoPackage
+  { unLicense :: Maybe Text.Text
+  , unLicenseFile :: Maybe Text.Text
+  }
+  deriving (Eq, Show)
+
+cargoPackageCodec :: TomlCodec CargoPackage
+cargoPackageCodec =
+  CargoPackage
+    <$> dioptional (Toml.text "license") .= unLicense
+    <*> dioptional (Toml.text "license-file") .= unLicenseFile
+
+-- |Representation of a Cargo.toml file. See
+-- [here](https://doc.rust-lang.org/cargo/reference/manifest.html)
+-- for a description of this format.
+newtype CargoToml = CargoToml
+  {cargoPackage :: CargoPackage}
+  deriving (Eq, Show)
+
+cargoTomlCodec :: TomlCodec CargoToml
+cargoTomlCodec = diwrap (Toml.table cargoPackageCodec "package")
+-- ^^ The above is a bit obscure. It's generating a TomlCodec CargoPackage and
+-- then using 'diwrap'/Coercible to make a TomlCodec CargoToml.  I can't use
+-- 'CargoToml <$>' because TomlCodec aliases (Codec a a) and only (Codec a)
+-- has a Functor instance, so I'd end up with a (Codec CargoPackage CargoToml).
+
+instance LicenseAnalyzeProject CargoProject where
+  licenseAnalyzeProject = analyzeLicenses . cargoToml
+
+-- |Analyze a Cargo.toml for license information. The format is documented
+-- (here)[https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields]
+analyzeLicenses :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m [LicenseResult]
+analyzeLicenses tomlPath = do
+  CargoPackage
+    { unLicenseFile = maybeLicenseFile
+    , unLicense = maybeLicense
+    } <-
+    cargoPackage <$> readContentsToml cargoTomlCodec tomlPath
+
+  -- The license-file field in Cargo.toml is relative to the dir of the
+  -- Cargo.toml file. Generate an absolute path to license-file.
+  let licensePathText = fmap prependTomlDir . parseRelFile . Text.unpack =<< maybeLicenseFile
```

Comment:
[suggestion]: Notably, we never actually use the `Text` version of this value, so we should consider parsing it into a `Filepath` instead of a `Text`.

It also deeply bothers me that `hlint` isn't telling you to use `toString` instead of `Text.unpack`, even though I know we've configured that rule.

Reply:
Also, we could probably parse it into a `Maybe (Path b File)` as part of the toml decoding.  It should work, since `parseRelFile` requires a monad with `MonadThrow`, like `Maybe`, and that's already on our datatype.

Reply:
I have made changes to make this a `Path Rel File`. Re: hlint, I think that the rules we had weren't permissive enough to catch my usage of `unpack` here. It was configured to match a usage like `Data.Text.unpack x` but wouldn't catch something like `Data.Text.unpack . foo`. I've made changes to .hlint.yaml that will hopefully remedy this.

Reply:
One other thing I didn't mention - having a `Maybe (Path Rel File)` for the license path means we could have a `Nothing` value for one of two reasons:

1. There was no `license-file` field.
2. There was a `license-file` field, but it could not be parsed.

Do we want to try to make this type more expressive so we can distinguish between these cases? Or is there some other part of the product that would catch this and is a more logical place to put it?

Reply:
Frustratingly, `tomland` doesn't actually support emitting custom errors.  I think it's probably best to use `Filepath` instead here, and just parse it under better error-reporting conditions.